src/packages/core/workspace/components/workspace-action/shared/workspace-action/workspace-action.element.ts

```diff
@@ -1,34 +1,53 @@
-import type { UmbWorkspaceAction } from './index.js';
+import type { UmbWorkspaceAction } from '../../../../index.js';
 import { UmbActionExecutedEvent } from '@umbraco-cms/backoffice/event';
 import { html, customElement, property, state } from '@umbraco-cms/backoffice/external/lit';
 import type { UUIButtonState } from '@umbraco-cms/backoffice/external/uui';
 import { UmbLitElement } from '@umbraco-cms/backoffice/lit-element';
 import type { ManifestWorkspaceAction } from '@umbraco-cms/backoffice/extension-registry';
 import { createExtensionApi } from '@umbraco-cms/backoffice/extension-api';
 
+import '../workspace-action-menu/index.js';
+
 @customElement('umb-workspace-action')
 export class UmbWorkspaceActionElement extends UmbLitElement {
+	#manifest?: ManifestWorkspaceAction;
+
 	@state()
 	private _buttonState?: UUIButtonState;
 
-	private _manifest?: ManifestWorkspaceAction;
 	@property({ type: Object, attribute: false })
 	public get manifest() {
-		return this._manifest;
+		return this.#manifest;
 	}
 	public set manifest(value: ManifestWorkspaceAction | undefined) {
 		if (!value) return;
-		const oldValue = this._manifest;
-		this._manifest = value;
-		if (oldValue !== this._manifest) {
+		const oldValue = this.#manifest;
+		this.#manifest = value;
+		if (oldValue !== this.#manifest) {
 			this.#createApi();
 			this.requestUpdate('manifest', oldValue);
 		}
 	}
 
+	@state()
```

Comment:
This can work for now, but for this to be as flexible as the overwrite feature, it should support overwrites of overwrites of overwrites.  And thats a bit complex, so for that to work, we should gather the aliases as part of the overwrite feature of the Extension Initialisers.

Another thing is, that this should not be calculated as a getter method on a state. Cause that will be triggered everytime there is a render for this element.

Instead do this as part of the setter of the manifest. As the Manifest is what determins the outcome of the calculation. Then thats can set the `_aliases` state to an array of strings. in this way it will just read te array when re-rendered and only do the calculation when the manifest is set. :-)

Reply:
Cool, thanks. Will do that.